src/index.js

```diff
@@ -1,42 +1,56 @@
-// ITERATION 1
-
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  console.log("Calculating subtotal, yey!");
 
   //... your code goes here
+  let price = product.querySelector(".price span");
+  let quantity = product.querySelector(".quantity input");
+
+  let produto = price.innerHTML * quantity.value;
+
+  let subtotal = product.querySelector(".subtotal span");
+
+  subtotal.innerHTML = produto;
+
+  return produto;
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
+ 
+  const singleProduct = document.querySelector(".product");
   updateSubtotal(singleProduct);
-  // end of test
 
   // ITERATION 2
-  //... your code goes here
+
+  let produtos = document.getElementsByClassName("product");
+
+  let totalValue = 0;
+  for (let i = 0; i < produtos.length; i++) {
+    totalValue += updateSubtotal(produtos[i]);
+  }
 
   // ITERATION 3
-  //... your code goes here
+
+  let total = document.querySelector("#total-value span");
+  total.innerHTML = totalValue;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  console.log('The target in remove is:', target.parentNode.parentNode);
+ target.parentNode.parentNode.remove();
```

Comment:
Every time you remove a product you need to update the Total Value, to do this you just need to call the function `calculateAll()` at the end of this function:

```
function removeProduct(event) {
  const target = event.currentTarget;
  console.log('The target in remove is:', target.parentNode.parentNode);
  target.parentNode.parentNode.remove()
  
  calculateAll()
}
```